src/core/hyprlock.cpp

```diff
@@ -793,6 +793,29 @@ void CHyprlock::renderOutput(const std::string& stringPort) {
     PMONITOR->sessionLockSurface->render();
 }
 
+void CHyprlock::startKeyRepeat(xkb_keysym_t sym) {
+    if (m_pKeyRepeatTimer) {
+        m_pKeyRepeatTimer->cancel();
+        m_pKeyRepeatTimer.reset();
+    }
+
+    m_pKeyRepeatTimer = addTimer(
+        /* initial delay until key repeat */ std::chrono::milliseconds(500), [sym](std::shared_ptr<CTimer> self, void* data) { g_pHyprlock->repeatKey(sym); }, nullptr);
+}
+
+void CHyprlock::repeatKey(xkb_keysym_t sym) {
+    handleKeySym(sym);
+
+    // This condition is for backspace and delete keys, but should also be ok for other keysyms since our buffer won't be empty anyways
+    if (bool CONTINUE = m_sPasswordState.passBuffer.length() > 0; CONTINUE)
+        m_pKeyRepeatTimer = addTimer(
+            /* key repeat delay */ std::chrono::milliseconds(50), [sym](std::shared_ptr<CTimer> self, void* data) { g_pHyprlock->repeatKey(sym); }, nullptr);
```

Comment:
you should get the values from wl_keyboard
![image](https://github.com/hyprwm/hyprlock/assets/43317083/2c238010-4a7d-4969-8d88-2e491aa1e1b6)